### DISCUSSION TITLE
Updated DSM today, got warnings about undeleted tmp files

### DISCUSSION THREAD

**Discussion by elisimpson** [007revad/Synology_HDD_db#279] [Q&A]
DSM update went okay and M.2 volume was fine after reboot, but got 2 of these warning messages in the log:

Synology_HDD_db update failed to delete tmp files.
Synology_HDD_db update failed to delete tmp files.

Is there anything I need to do to clean up these files?

**Comment by 007revad** [comment]
Interesting that you got 2 of those. That means the script failed to delete the previously downloaded .tar.gz file and the extracted tmp files, as well as the current .tar.gz file and the extracted tmp files. Which means it also failed to delete it's tmp files the last time it updated.

DSM empties the tmp folder when the NAS reboots, so the files will eventually be deleted by DSM. But if the script ran at boot and you got those log entries then DSM didn't delete the temporary files.

What do the following commands return:
```
ls /tmp | grep -E 'Synology_*'
ls /tmp/Synology_HDD_db-3.5.89
```

**Comment by elisimpson** [comment]
Nothing to both. All the files in /tmp are from the time I did the upgrade (today) except for 2:

drivedb.db (143k, dated 8/23/2023)
synodrivedb.db (92k, dated 8/10/2023)

Also I see that a couple of files got added to the dir I run the script from: syno_hdd_db_CHANGES.txt and syno_hdd_vendor_ids.txt

**Comment by 007revad** [comment]
By the date of those old drivedb.db and synodrivedb.db files it looks like DSM does not empty /tmp at boot. I know some Linux distros don't, but others do.

If you haven't noticed the syno_hdd_db_CHANGES.txt and syno_hdd_vendor_ids.txt files until now you must have been using a fairly old version.
- syno_hdd_vendor_ids.txt was added 4 months ago. 
- CHANGES.txt was renamed to <scriptname>_CHANGES.txt 3 months ago.

The files in /tmp failing to be deleted may have been a bug in the version you had before you updated. It looks like the bug thought the files failed to delete when they actually did delete.

**Comment by elisimpson** [comment]
I had CHANGES.txt (not _CHANGES.txt) but not syno_hdd_vendor_ids.txt. As I typically only get an update when rebooting it makes sense that the version was fairly old. I'll keep an eye out next time DSM updates (or I need to reboot). Thanks for your help!
